derive.go

```diff
@@ -3,13 +3,16 @@ package schnorrkel
 import (
 	"crypto/rand"
 	"errors"
-
 	"github.com/gtank/merlin"
 	r255 "github.com/gtank/ristretto255"
 )
 
 const ChainCodeLength = 32
 
+var (
+	ErrDeriveHardKeyType = errors.New("Error deriving hard key type, DerivableKey must be a SecretKey")
```

Comment:
please change `Error deriving hard key type` to `failed to derive hard key type`